server/rest/src/main/java/org/infinispan/rest/resources/CacheResourceV2.java

```diff
@@ -241,6 +252,21 @@ public Invocations getInvocations() {
       }, invocationHelper.getExecutor());
    }
 
+   private CompletionStage<RestResponse> cacheListen(RestRequest request) {
+      MediaType accept = negotiateMediaType(request, APPLICATION_JSON, TEXT_PLAIN);
+      String cacheName = request.variables().get("cacheName");
+      boolean includeCurrentState = Boolean.parseBoolean(request.getParameter("includeCurrentState"));
+      RestCacheManager<Object> restCacheManager = invocationHelper.getRestCacheManager();
+      if (!restCacheManager.cacheExists(cacheName))
+         return notFoundResponseFuture();
+      Cache<?, ?> cache = restCacheManager.getCache(cacheName, accept, accept, request);
+      BaseCacheListener listener = includeCurrentState ? new StatefulCacheListener(cache) : new StatelessCacheListener(cache);
+      cache.addListener(listener);
```

Comment:
Do we need to perform cleanup for listeners once the event stream has been closed? Without this we will add a new listener to the cache everytime a new SSE request is received which could be probelematic over time.

Reply:
BlockHound is complaining about the call to `addListener`. We should probably use `addListenerAsync` here and return the response when complete.

Reply:
I've changed the code to use `addListenerAsync`